src/librustc_typeck/check/method/suggest.rs

```diff
@@ -536,18 +536,74 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
 
                 if !unsatisfied_predicates.is_empty() {
+                    let mut bound_spans = vec![];
                     let mut bound_list = unsatisfied_predicates
                         .iter()
-                        .map(|p| format!("`{} : {}`", p.self_ty(), p.print_only_trait_path()))
+                        .filter_map(|p| {
+                            let self_ty = p.self_ty();
+                            match &self_ty.kind {
+                                ty::Adt(def, _) => {
+                                    bound_spans.push((
+                                        self.tcx
+                                            .sess
+                                            .source_map()
+                                            .def_span(self.tcx.def_span(def.did)),
+                                        format!(
+                                            "the method `{}` exists but this type doesn't satisfy \
+                                             the bound `{}: {}`",
+                                            item_name,
+                                            p.self_ty(),
+                                            p.print_only_trait_path()
+                                        ),
+                                    ));
+                                    None
+                                }
+                                ty::Dynamic(preds, _) => {
+                                    for pred in *preds.skip_binder() {
+                                        match pred {
+                                            ty::ExistentialPredicate::Trait(tr) => bound_spans
+                                                .push((
+                                                    self.tcx
+                                                        .sess
+                                                        .source_map()
+                                                        .def_span(self.tcx.def_span(tr.def_id)),
+                                                    format!(
+                                                        "the method `{}` exists but this trait \
```

Comment:
We're talking about a trait object type here not satisfying a bound, no? Not a trait?

Reply:
Correct. I wanted to push out the PR asap for feedback, but the wording is still in flux. I wanted to verify I could get all the pieces in place first.

Reply:
s/trait/trait object type/ basically

Reply:
Changed the output in a way that it no longer mentions trait object type.